wifi: Add status enumerations

krish2718 opened the pull request.

krish2718 committed:
wifi: Add an enum for connect result status

This can be used to interpret the "status" in the connect result event.
To start with a few common reasons are enumerated but can be extended in
the future.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
wifi: Add an enum for disconnect reasons

This can be used to interpret the "status" in the disconnect result event.
To start with a few common reasons are enumerated but can be extended in
the future.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

jukkar commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -321,6 +321,36 @@ struct wifi_connect_req_params {
+
 /** Generic Wi-Fi status for commands and events */
 struct wifi_status {
 	int status;
```
If those conn/disconn status are mapped to this status field, I think it would make sense to have a union here so that it is explicit. So something like
```
struct wifi_status {
    union {
        int status;
        enum wifi_conn_status conn_status;
        enum wifi_disconn_reason disconn_reason;
    };
};
```

krish2718 committed:
wifi: Add the enums to the status

This helps to demonstrate that the enumerations have to be used to
interpret the status.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

kartben reviewed (changes requested):
Please consider this change for the brief description to actually be brief + spelling
Thanks!
